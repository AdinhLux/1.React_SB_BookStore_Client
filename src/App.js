// We need to import React to make this const a component
import React from 'react';

// Creating a dummy component
const App = () => {
    return <div>My first component</div>;
}

// Export to outside as 'App'
export default App;