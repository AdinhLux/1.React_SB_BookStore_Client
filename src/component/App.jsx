// We need to import React to make this const a component
import React from "react";

import axios from "axios";
import baseUrl from "../config";

import Layout from "./layout/Layout";

// Creating a dummy component
const App = () => {
  // Endpoint of Book to get all books
  axios(`${baseUrl}/api/v1/books`).then((books) => {
    console.log(books);
  });

  return (
    <Layout>
      <div>My first component</div>
    </Layout>
  );
};

// Export to outside as 'App'
export default App;
