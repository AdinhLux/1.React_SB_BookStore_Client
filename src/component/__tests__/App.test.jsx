import React from "react";
import { render } from "@testing-library/react";

// Import your component for testing
import App from "./App";

describe("App component", () => {
  it("should render app with error", () => {
    // asFragment' is a method that returns a DocumentFragment of our rendered component.
    const { asFragment } = render(<App />);

    // Testing with Jest snapshot
    // Snapshots are generally testing the initial rendering of a component
    //
    // toMatchSnapshot() is a function to compare the current snapshot with the newly generated output.
    expect(asFragment()).toMatchSnapshot();
  });
});
