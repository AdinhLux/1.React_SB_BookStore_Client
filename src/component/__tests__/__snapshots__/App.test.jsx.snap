// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render app with error 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <header
        class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
      >
        <div
          class="MuiToolbar-root MuiToolbar-dense MuiToolbar-gutters"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
          >
            Book Store
          </h6>
        </div>
      </header>
    </div>
    (mt = margin top, ml = margin left)
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <div
        class="MuiBox-root MuiBox-root-10 makeStyles-bookContainer-4"
      >
        <div
          class="MuiBox-root MuiBox-root-11 makeStyles-bookFilter-5"
        >
          <div
            class="MuiPaper-root makeStyles-bookFilterPaper-6 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <p
              class="MuiTypography-root MuiTypography-body1"
            >
              Search Books Filter
            </p>
            <div
              class="MuiBox-root MuiBox-root-12"
            >
              <div
                class="MuiFormControl-root MuiTextField-root"
                data-testid="book-title-input"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
                  data-shrink="false"
                  for="book-search"
                  id="book-search-label"
                >
                  Enter book title
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    id="book-search"
                    placeholder="Enter book title"
                    type="text"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-13 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-15"
                    >
                      <span>
                        Enter book title
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Search
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiBox-root MuiBox-root-17 makeStyles-bookList-7"
        >
          <div
            class="MuiBox-root MuiBox-root-18"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-pulse"
              data-testid="book-loader"
              style="width: 80%; height: 200px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
